modules/middleware/src/buildAggregations/index.js

```diff
@@ -97,10 +98,18 @@ export default function ({
   const aggs = Object.entries(graphqlFields).reduce((aggregations, [fieldKey, graphqlField]) => {
     const field = fieldKey.replace(/__/g, '.');
     const nestedPaths = getNestedPathsInField({ field, nestedFields });
+    const contentsFiltered = (normalizedSqon?.content || []).filter((c) =>
+      aggregationsFilterThemselves
+        ? c.content?.field?.startsWith(nestedPaths)
+        : c.content?.field?.startsWith(nestedPaths) && c.content?.field !== field,
+    );
+    const termFilters = contentsFiltered.map((filter) => opSwitch({ nestedFields: [], filter }));
```

Comment:
Can we remove the consideration for ordering of filters (ie. startsWith) . We should assume all filters inside a pivot operation are meant to be nested in ES.

Reply:
Do yuo mean doind something like this instead?:
```
const contentsFiltered = (normalizedSqon?.content || []).filter((c) =>
      aggregationsFilterThemselves && c.content?.field !== field,
 );
```

Reply:
Having read through this more in depth and tested it out, the original code makes sense - I was confused. This works as written.